staging/src/k8s.io/cli-runtime/pkg/resource/visitor.go

```diff
@@ -612,7 +612,7 @@ func SetNamespace(namespace string) VisitorFunc {
 		}
 		if len(info.Namespace) == 0 {
 			info.Namespace = namespace
-			UpdateObjectNamespace(info, nil)
+			return UpdateObjectNamespace(info, nil)
```

Comment:
I've added this `return` to return the error and there is an error indeed! Turns out one of the tests is failing because this `info` is a `PodList` with two `Pod`s but this function doesn't handle lists. I don't know if it should or not, but clearly it's some sort of a bug. @seans3, could you have a look? Or perhaps you know who might be the right person to figure this out? Pinging you since the bot assigned you :)